### DISCUSSION TITLE
Scheduler is going haywire

### DISCUSSION THREAD

**Discussion by SunDevil311** [d4rken-org/sdmaid-se#519] [Q&A]
I'm not sure what prompted the behavior, but I'd say about 2-3 days ago (roughly) I noticed that Scheduler was executing multiple times throughout the day.

Though easily enough cancelled, the accessibility service feature would basically lock everything up until "Cancel" was pressed.

The problem appears to occur, at a minimum, every time I restart or power off / on the device.

The only major change I can think of during this time was a T-Mobile OTA security patch update. For now, I've simply deleted the existing schedule.

Model: SM-G981U (Galaxy S20 5G)
OneUI Version: 5.1
Android Version: 13

**Comment by d4rken** [comment]
Which SD Maid update are you on?

There might be two issues here.

1: The accessibility service should not be used by the scheduler, unless this is explicitly enabled via settings. This should be off by default. Did you enable it?

2: My first thought was that some kind of retry-logic keeps restarting the scheduled task because it failed before, but I just checked the code and there is no retry logic. 

Can you keep a debug log running and reproduce this behaviour?

**Reply by SunDevil311** [reply]
Edit: using version 0.10.7-beta0 from the Play Store.

My apologies, I did in fact enable the Accessibility toggle in Settings. I clearly misread and/or misunderstood the description, as I thought that I was simply enabling use of the Accessibility Service for automated cleaning (which isn't necessary beyond granting the app its initially requested permissions, unless I'm mistaken).

Thanks for clearing that up so quickly!
Reading comprehension ftl lol

Would you like me to attempt to reproduce for debugging purposes? I'm not sure whether that behavior would be expected upon enabling the toggle.

**Reply by d4rken** [reply]
Not sure if we are talking about the same setting.

You can and should enable the accessibility service in general, but if the app cleaner task was launched by the scheduler, then the accessibility service should not be used, unless this setting is enabled in the **scheduler** settings. If that is not the case, please record a log.

[Screenshot_20230722-221858.png](https://github.com/d4rken-org/sdmaid-se/assets/1439229/3a0a92ec-4bb3-4157-8e0e-82d25ce2fe9f)

**Reply by SunDevil311** [reply]
Yes, we are talking about the same thing. For some reason I toggled the setting in your screenshot to "on."

**Reply by d4rken** [reply]
Okay then that is working as expected. The remaining issue would be the unexpected scheduler execution. If you can somehow find out if there is a pattern or keep the debug log running to record the random events, then this might give me some clues.
